board.pl

```diff
@@ -53,37 +55,78 @@
         axis(A, direction(W,V))
     ).
 
-% TODO
-% slot(Num, Letter)
+letter_to_row(L, I):-
+    char_code(L, LCode),
+    I is LCode - 64.
 
-% TODO
-% % matches slot's color, depends on board state
-% % slot_color(Board, Slot, Color)
-% slot_color(Board, slot(Num, Letter), color(Color)).
+% matches slot content, depends on board state
+slot_by_index(BoardState, RowIndex, ColIndex, Slot):-
```

Comment:
```suggestion
slot_by_index(BoardState, RowIndex, ColIndex, SlotType):-
```

Reply:
Whoa yeah that's better, gonna have to change just about everything in the file XD